py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py

```diff
@@ -239,3 +240,299 @@ def _add_layer(
         input_val = clamp_max_layer.get_output(0)
 
     return input_val
+
+
+def add(
+    network: TRTNetwork,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    lhs_val: TRTTensor,
+    rhs_val: TRTTensor,
+) -> TRTTensor:
+    if lhs_val.dtype == trt.int8 or lhs_val.dtype == trt.int32:
+        lhs_val = cast_trt_tensor(network, lhs_val, trt.float32, name)
+
+    if rhs_val.dtype == trt.int8 or rhs_val.dtype == trt.int32:
+        rhs_val = cast_trt_tensor(network, rhs_val, trt.float32, name)
```

Comment:
The Dynamo error encountered is related to this change, because the type for inputs to all of the `elementwise` layers is generally `Union[TRTTensor, int, float]`, since we can encounter cases where floats get added to `TRTTensor`s for example. As a result, the code here fails, since it tries to access the `dtype` of a Python float.

Reply:
The above code seems to show up a lot. Consider adding a function in `converter_utils.py`, which performs this check and cast automatically. Something like:

```python
def trt_cast_int_to_float(tensor: TRTTensor):
    if tensor.dtype == trt.int8 or tensor.dtype == trt.int32:
        return cast_trt_tensor(network, tensor, trt.float32, name)

    return tensor
```

Reply:
Then, you can refactor the code like this, for instance:
```python
    if lhs_val.dtype == trt.int8 or lhs_val.dtype == trt.int32:
        lhs_val = cast_trt_tensor(network, lhs_val, trt.float32, name)

##### REPLACED BY

    if isinstance(lhs_tensor, TRTTensor):
        lhs_tensor = trt_cast_int_to_float(lhs_tensor)
```